src/logger/index.ts

```diff
@@ -70,34 +72,68 @@ const createBaseLogger = (config: LogConfig) => {
   });
 };
 
+// TODO: Move to its own file so internalAsyncLocalStorage is not available and rename
+let internalAsyncLocalStorage: AsyncLocalStorage<LogContext>;
```

Comment:
There shouldn't be any risks of having a singleton here right?

Reply:
No, it should be OK.